File: frontend/src/service/FavoriteService.js
import {GET} from '../utils/Axios';
import {url} from '../config/url';

export const listFavoriteByPage = (data, callback) => {
    const request = `${url}/Favorite/listFavoriteByPage`;
    GET(request, data, callback);
};

export const addFavorite = (data, callback) => {
    const request = `${url}/Favorite/addFavorite`;
    GET(request, data, callback);
};

export const deleteFavorite = (data, callback) => {
    const request = `${url}/Favorite/deleteFavorite`;
    GET(request, data, callback);
};

export const checkFavorite = (data, callback) => {
    const request = `${url}/Favorite/checkFavorite`;
    GET(request, data, callback);
};

export const deleteAllFavorite = (data, callback) => {
    const request = `${url}/Favorite/deleteAllFavorite`;
    GET(request, data, callback);
};

File: backend_test/src/main/java/com/backend/passthemon/service/FavoriteService.java
package com.backend.passthemon.service;

import com.backend.passthemon.dto.FavoriteDto;
import com.backend.passthemon.entity.Favorite;
import com.backend.passthemon.entity.Goods;
import com.backend.passthemon.entity.User;
import com.backend.passthemon.utils.msgutils.Msg;

import java.util.List;

public interface FavoriteService {
    List<FavoriteDto> listFavoriteByPage(Integer fetchPage, Integer userId);
    Favorite addFavorite(Integer userId, Integer goodsId);
    Favorite deleteFavorite(Integer userId, Integer goodsId);
    Favorite deleteAllFavorite(Integer userId);
    Favorite checkFavorite(Integer userId, Integer goodsId);
}
